index.js

```diff
@@ -169,7 +169,12 @@ module.exports = function makeFetch (opts = {}) {
         let statusCode = 200
 
         if (resolved.type === 'directory') {
-          const files = await archive.readdir(finalPath)
+          const files = await Promise.all(await archive.readdir(finalPath).then(files => files.map(async fileName => {
```

Comment:
This is very verbose but I couldn't think of a shorter way to do everything asynchronously

Reply:
Check out the `includeStats` method for readdir: https://github.com/Wapikoni-Technique/Natakanu/blob/d8886cce51badc4b5395bcb1d4db964344d996a7/app/core/Account.js#L68

When you invoke it, what you get back is an array of objects with `{stat, name}`. That way you can have something like

```js
const stats = await archive.readdir(finalPath)

const files = stats.map({stat, name} => stat.isDirectory() ? `${name}/` : name)
```